### PR TITLE
Add NonEmptyStringScalar

### PR DESCRIPTION
**Pull request by PascalSenn**
Adds NonEmptyStringScalar

### CODE REVIEW

File: src/HotChocolate/Core/src/Types.Scalars/ThrowHelper.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+namespace HotChocolate.Types.Scalars
+{
+    internal static class ThrowHelper
+    {
+        public static SerializationException NonEmptyStringType_ParseLiteral_IsEmpty(IType type)
+        {
+            return new SerializationException(
+                ErrorBuilder.New()
```
Comment (fredericbirke) [inline src/HotChocolate/Core/src/Types.Scalars/ThrowHelper.cs:8]: Add HC0002 error code

File: src/HotChocolate/Core/src/Types.Scalars/ThrowHelper.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                    .SetMessage(ScalarResources.NonEmptyStringType_IsEmpty_ParseLiteral)
+                    .SetExtension("actualType", "NonEmptyString")
+                    .Build(),
+                type);
+        }
+
+        public static SerializationException NonEmptyStringType_ParseValue_IsEmpty(IType type)
+        {
+            return new SerializationException(
+                ErrorBuilder.New()
```
Comment (fredericbirke) [inline src/HotChocolate/Core/src/Types.Scalars/ThrowHelper.cs:18]: Add HC0001 Error Code

**Review by fredericbirke — CHANGES_REQUESTED** [review]

**Review by fredericbirke — APPROVED** [review]
LGTM

File: src/HotChocolate/Core/test/Types.Scalars.Tests/NonEmptyStringTypeTests.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        [Theory]
+        [InlineData(typeof(EnumValueNode), TestEnum.Foo, false)]
+        [InlineData(typeof(FloatValueNode), 1d, false)]
+        [InlineData(typeof(IntValueNode), 1, false)]
+        [InlineData(typeof(BooleanValueNode), true, false)]
+        [InlineData(typeof(StringValueNode), "", false)]
+        [InlineData(typeof(StringValueNode), "foo", true)]
+        [InlineData(typeof(NullValueNode), null, true)]
+        public void Test_IsInstanceOfTypeValueNode(Type type, object value, bool expected)
```
[the remark this answers is not in the document]
Comment (PascalSenn) [reply src/HotChocolate/Core/test/Types.Scalars.Tests/NonEmptyStringTypeTests.cs]: done

**Pull request merged**